### DISCUSSION TITLE
How to use pdf.js-dist in next js

### DISCUSSION THREAD

**Discussion by Harsh-Sharma25** [mozilla/pdf.js#17523] [Q&A]
How to configure pdf.js-dist to be able to run in next js
next-js version:- 12.3.1

This is the error I get:

./node_modules/pdfjs-dist/build/pdf.mjs
Module parse failed: The top-level-await experiment is not enabled (set experiments.topLevelAwait: true to enabled it)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders

Error: The top-level-await experiment is not enabled (set experiments.topLevelAwait: true to enabled it)

Even after enabling this in next-js config experiments.topLevelAwait. I still get the same error.

**Comment by AnilNallaTR** [comment]
Hello, is there any solution for this issue? I am getting below error when i upgrade pdfjs-dist version to 4.2.67 and i have added below script in both webconfigure.config.ts and angular.json but still am getting issue

angular.json:
                         "test": {
					"builder": "@angular-builders/custom-webpack:karma",
					"options": {
						"customWebpackConfig": {
							"path": "./webpack.config.ts"
							"mergeStrategies": { "experiments": "replace" }
						},
				}
```

webpack.config.ts :


	config.experiments = config.experiments || {}; 
	config.experiments.topLevelAwait = true;


./node_modules/pdfjs-dist/build/pdf.mjs - Error: Module parse failed: The top-level-await experiment is not enabled (set experiments.topLevelAwait: true to enabled it)
File was processed with these loaders:
 * ./node_modules/@angular-devkit/build-angular/src/babel/webpack-loader.js
 * ./node_modules/source-map-loader/dist/cjs.js
You may need an additional loader to handle the result of these loaders.
Error: The top-level-await experiment is not enabled (set experiments.topLevelAwait: true to enabled it)
